### ISSUE TITLE
Consider snapping modes

### ISSUE THREAD

**Issue by davidgilbertson** [mapbox/mapbox-gl-draw#865]
Hello there map friends. I've written some custom modes so the drawing tools snap to right angles and other features.

https://github.com/davidgilbertson/draw-geoson (and https://drawgeojson.firebaseapp.com/)

It's just the start of a larger project for me, but if you'd like to include the logic in this repo that'd be cool.

**Comment by vc-void** [comment]
Pretty awesome, any chance you can update the repo link?

**Comment by davidgilbertson** [comment]
That repo grew into a commercial product and is no longer open source, sorry about that.

**Comment by mhsattarian** [comment]
Hi @davidgilbertson, I found your example the only one currently working properly, I collected the code from your example in a [Stackblitz project](https://stackblitz.com/edit/mapboxgl-draw-snap-react?file=src/Map/snapModes/SnapPolygonMode.js).
I hope It's ok since the code in your example was already public online. If not I will remove it.

for anyone looking for a snapping mode for `mapbox-gl-draw`, I've also created a snapping mode, which can snap to segments (in addition to vertices as in @davidgilbertson example). will publish that soon.
